src/main/java/uk/gov/companieshouse/api/testdata/service/impl/TestDataServiceImpl.java

```
@@ -185,4 +189,32 @@ public boolean deleteUserData(String userId) {
         }
         return this.userService.delete(userId);
     }
+
+    @Override
+    public IdentityData createIdentityData(IdentitySpec identitySpec) throws DataException {
+        if (identitySpec.getUserId() == null || identitySpec.getUserId().isEmpty()) {
+            throw new DataException("User Id is required to create an identity");
+        }
+        if (identitySpec.getEmail() == null || identitySpec.getEmail().isEmpty()) {
+            throw new DataException("Email is required to create an identity");
+        }
+        if (identitySpec.getVerificationSource() == null
+                || identitySpec.getVerificationSource().isEmpty()) {
+            throw new DataException("Verification source is required to create an identity");
+        }
+        try {
+            return identityService.create(identitySpec);
+        } catch (Exception ex) {
+            throw new DataException("Error creating identity", ex);
+        }
+    }
+
+    @Override
+    public boolean deleteIdentityData(String identityId) {
+        try {
+            return identityService.delete(identityId);
+        } catch (Exception ex) {
+            return false;
+        }
```

Should we throw the exception rather than "eating it"

---

updated as suggested